articles/list-transformations-js/index.md

```diff
@@ -0,0 +1,357 @@
+4. Using the square notation adds another object at index 1.
+
+### Accessing nested data structures
+
+We can access the first item of the nested array `brands` above using the bracket notation as shown below:
+```javascript
+var car1 = cars.brands[0];
+
+console.log(car1); //{ name: "Ford", origin: "USA"}
+```
+Dot notation can't be used above as the indices of an array are numeric and the dot notation only accepts strings.
+
+We can also access the `name` property of the first object of the `brands` array using the dot notation.
+```javascript
+var car1Name = cars.brands[0].name;
+
+console.log(car1Name); //Ford
+```
+### Looping through nested arrays using a `for()` loop
+
+Let us loop through all the objects in the nested array `brands` we created earlier.
+```javascript
+function allCars(brands){
+	for (let i = 0; i < brands.length; i++){
+    console.log(brands[i])
+	};
+};
+
+allCars(cars.brands);
+//{ name: 'Ford', origin: 'USA' }
+//{ name: 'Toyota', origin: 'japan
+```
+In the code above:
+1. We declare a function `allCars()` with a parameter `brands`.
+2. Inside it we create a [`for()` loop](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for) that loops through `brands` and logs them.
+3. We call the function `allCars()`and pass an argument `cars.brands`.
+
+### Looping through nested objects using a `for()` loop
+
+We can get common properties in nested objects using a for loop. Let us log all the `name` properties of the nested brand objects.
+
+```javascript
+function names(brands){
+    for (var i = 0; i < brands.length; i++){
+        console.log(brands[i].name)
+    };
+};
+
+names(cars.brands); 
+//Ford
+//Toyota
+```
+In the above code:
+1. We declare a function `names()` with a parameter `brands`.
+2. We create a for loop that logs each `name` property of the nested objects.
+3. We then call the `names()` function and pass an argument `cars.brands`.
+
+### Looping through nested objects properties using a `for ...in` loop
+
+If we want to get all the properties of a nested object, we can use a [`for ...in`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...in) loop. Let's loop through all the properties of the nested `brands` array.
+
+```javascript
+function everything (brands){
+    for(var i =0; i< brands.length; i++){
+        for (var property in brands[i]){
+            console.log(brands[i][property])
+        }
+    }
+};
+
+everything(cars.brands); 
+//Ford
+//USA
+//Toyota
+//Japan
+```
+In the above code:
+1. We declare a function `everything()` with a parameter `brands`.
+2. Inside it, we create a [`for()` loop](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for) that loops through all the items in the `brands` array.
+3. Inside it, we create a [`for ...in`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...in) loop that loops through all the object properties.
```

Comment:
Line 131 & 132
"Inside it" is used twice.
To improve readability, consider mentionin what you are reffering to using "it" instead of "it".

"Inside `everything()`, we create..."
"Inside the `for()` loop, we create..."